lib/spack/spack/bootstrap/_common.py

```diff
@@ -213,15 +213,19 @@ def _root_spec(spec_str: str) -> str:
     Args:
         spec_str: spec to be bootstrapped. Must be without compiler and target.
     """
-    # Add a compiler requirement to the root spec.
+    # Add a compiler and platform requirement to the root spec.
     platform = str(spack.platforms.host())
+
     if platform == "darwin":
         spec_str += " %apple-clang"
+    elif platform == "windows":
+        spec_str += " %msvc"
     elif platform == "linux":
         spec_str += " %gcc"
     elif platform == "freebsd":
         spec_str += " %clang"
-
+    # Note: this may be too restrictive for freebsd
+    spec_str += f" platform={platform}"
```

Comment:
@haampie Is this too restrictive for freebsd?

Reply:
no